test/unit/scan/test_scan_policies.py

```diff
@@ -7,14 +7,207 @@
 # License https://www.backblaze.com/using_b2_code.html
 #
 ######################################################################
+from pathlib import Path
+import platform
 import re
-
+from unittest.mock import MagicMock
+from b2sdk.scan.folder import LocalFolder
+from b2sdk.utils import fix_windows_path_limit
 import pytest
 
 from apiver_deps import ScanPoliciesManager
 from apiver_deps_exception import InvalidArgument
 
 
+class TestFolderTraversal:
+    def test_simple_folder(self, tmp_path):
+        d = tmp_path / "dir"
+        d.mkdir()
+        (d / "file1.txt").write_text("content1")
+        (d / "file2.txt").write_text("content2")
+        (d / "file3.txt").write_text("content3")
+
+        folder = LocalFolder(str(d))
```

Comment:
generally we should use Pathlib everywhere. `b2sdk.v2` might use string paths in some places, but it's a legacy problem and I think switching to pathlib in `v3` is a good direction (though maybe lets not do it in this PR).